File: 2/logical.py
#!/usr/bin/python2.7 -u
# logical operators: || && ! and or not

print "Logical Operator test"
a = 11
b = 10

if a > b and b < 100:
    answer = "yes"
else:
    answer = "no"
print answer

c = 0

if a and c:
    print "Nothing to see here..."
else:
    print "Didn't see anything..."

if not c:
    print "But we will see this!"

if a and c or b:
    print "and has lower precedence than ||"

if a and c or c:
    print "&& has higher precedence than or"

if a and not c:
    print "not not not and and and"

File: 2/logical.pl
#!/usr/bin/perl -w
# logical operators: || && ! and or not

print "Logical Operator test\n";
$a = 11;
$b = 10;

if ($a > $b && $b < 100) {
	$answer = "yes";
} else {
	$answer = "no";
}
print "$answer\n";

$c = 0;

if ($a && $c) {
	print "Nothing to see here...\n";
} else {
	print "Didn't see anything...\n";
}

if (!$c) {
	print "But we will see this!\n";
}

if ($a and $c || $b) {
	print "and has lower precedence than ||\n";
}

if ($a && $c or $c) {
	print "&& has higher precedence than or\n";
}

if ($a and not $c) {
	print "not not not and and and\n";
}